src/Modules/ModulesGraded.jl

```diff
@@ -27,13 +27,13 @@ julia> R, (x,y) = graded_polynomial_ring(QQ, ["x", "y"])
 (Graded multivariate polynomial ring in 2 variables over QQ, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}[x, y])
 
 julia> graded_free_module(R,3)
-Graded free module R^3([0]) of rank 3 over R
+Graded free module Graded multivariate polynomial ring^3([0]) of rank 3 over graded multivariate polynomial ring
```

Comment:
I think this will be objected to by @jankoboehm and @wdecker and should be fixed. Probably it has to do with switches from `compact` to `supercompact` somewhere? Probably only the old method's signature (of the one which was looking up the rings to fish out their names) has to be adjusted.

Reply:
For now I made it set both `compact` and `supercompact`, but this will be the last change I'll make to this PR, from here on the algebraic geometry group should take over, as outlined in the description of this PR.